build.xml

```diff
@@ -78,24 +80,23 @@
   <!-- Create jar file, exclude tests  -->
   <target name="dist-jar" depends="build-jar">
     <mkdir dir="${dir-dist}" />
-    <jar destfile="${dir-dist}/${ant.project.name}-${version}.jar" basedir="${dir-bin}">
-      <exclude name="**/test/**" />
-    </jar>
+    <jar destfile="${dir-dist}/${ant.project.name}-${version}.jar" basedir="${dir-bin}" />
   </target>
 
   <!-- Create jar file, include the tests ONLY  -->
   <target name="dist-tests" depends="build-tests">
     <mkdir dir="${dir-dist}" />
-    <jar destfile="${dir-dist}/${ant.project.name}-${version}-tests.jar" basedir="${dir-bin}">
-      <include name="**/test/**" />
-    </jar>
+    <jar destfile="${dir-dist}/${ant.project.name}-${version}-tests.jar" basedir="${dir-bin}" />
   </target>
 
   <!-- Run tests using built artifacts -->
-  <target name="run-tests" depends="dist-jar, dist-tests">
+  <target name="run-tests" depends="dist-tests">
     <junit printsummary="yes" haltonfailure="yes">
+      <formatter type="plain" />
       <classpath refid="classpath-tests" />
-      <!-- Add tests here -->
+      <batchtest fork="yes" todir="${dir-test-report}">
+        <fileset dir="${dir-src-test}" includes="**/*Test.java"/>
+      </batchtest>
```

Comment:
I like it slightly better to list the single test classes here (using the `test` target), at least as long as there are only a few of them:

    <test name="com.suse.saltstack.netapi.client.FoobarTest" todir="${dir-test-results}">
        <formatter type="plain" />
    </test>